R/register.R

```diff
@@ -136,6 +136,18 @@ get_registered_functions <- function(decorations, tag, quiet = FALSE) {
   }
 
   out <- decorations[decorations$decoration == tag, ]
+
+  if(any(!decorations$decoration %in% c("cpp11::register", "cpp11::init"))) {
+    lines <- decorations$line[which(!decorations$decoration %in% c("cpp11::register", "cpp11::init"))]
+    if(length(lines) > 1) {
+      stop(call. = FALSE, paste0("Can't capture cpp11 decorators on lines ",
```

Comment:
Maybe the error should be something like 

```
cpp11 attributes must be either `cpp11::register` or `cpp11::init`:
- Invalid attribute [[cpp11::foobar]] on line 123 in file xyz.
- Invalid attribute [[cpp11::baz]] on line 456 in file abc.
```

Reply:
Oh that looks great!
Is the best way to make multi-line error messages by using a combination of `paste()` statements with `sep = "\n"` ? Or would this be a `message()` following by an empty `stop()` ?

Reply:
You can actually use [glue](https://github.com/tidyverse/glue) to do this a little more nicely, something like this

```r
bad_lines <- glue::glue_collapse(
  glue::glue("- Invalid attribute `{names}` on line {line} in file '{file}'."),
  "\n"
)

msg <- glue::glue("
  cpp11 attributes must be either `cpp11::register` or `cpp11::init`:
  {bad_lines}
  ")
stop(msg, call. = FALSE)
```

Reply:
Oh nice!